src/sonic-config-engine/minigraph.py

```diff
@@ -157,11 +161,94 @@ def parse_png(png, hname):
                             elif node.tag == str(QName(ns, "EndDevice")):
                                 mgmt_dev = node.text
 
-    return (neighbors, devices, console_dev, console_port, mgmt_dev, mgmt_port, port_speeds, console_ports)
+    return (neighbors, devices, console_dev, console_port, mgmt_dev, mgmt_port, port_speeds, console_ports, device_neighbors)
 
+def parse_asic_png(png, asic_name, hostname):
+    neighbors = {}
+    devices = {}
+    port_speeds = {}
+    device_neighbors = []
```

Comment:
do not understand why we still have device_neighbors here since it it asic png?

Reply:
When we get the list of "Devices" in the minigraph, we get list of all devices present, but we don't want all devices to be a part of the "neighbors" list. This list will keep track of neighbor device names. While populating the "DEVICE_NEIGHBORS_METADATA" table, we will populate information of a device only if that device is a neighbor of the ASIC.

Reply:
removed device_neighbors